mlflow/tracking/client.py

```diff
@@ -963,6 +965,47 @@ def log_text(self, run_id, text, artifact_file):
             with open(tmp_path, "w") as f:
                 f.write(text)
 
+    def log_dict(self, run_id, dct, artifact_file):
```

Comment:
```suggestion
    def log_dict(self, run_id, dictionary, artifact_file):
```

BTW, I was wondering if it'd make sense to just use a JSON format by default if no file extension is specified, but I guess our artifact viewer won't render it particularly well, so maybe it's fine to be restrictive at first since at least for workflows we're aware of, the user usually intends to log the dict in JSON or YAML format.